File: src/DepoWise.Application/Ui/Menu.cs
using DepoWise.Application.Security;

namespace DepoWise.Application.Ui;

public sealed record MenuItem(string Key, string Label);

/// <summary>
/// Menü kurucu — deny-by-default. Yalnız oturumun okuma (CanSeeMenu) yetkisi olan modülleri
/// üretir. Web ve masaüstü AYNI sonucu kullanır (tek doğru kaynak: AppModules + AccessControl).
/// </summary>
public static class MenuBuilder
{
    public static IReadOnlyList<MenuItem> Build(SessionContext session)
    {
        var items = new List<MenuItem>();
        foreach (var (key, label) in AppModules.All)
        {
            if (AccessControl.CanSeeMenu(session, key))
                items.Add(new MenuItem(key, label));
        }
        return items;
    }
}

File: apps/web/src/lib/ui/menu.ts
import { canSeeMenu, type Session } from "../security/permissions.ts";
import { APP_MODULES, type ModuleDef } from "./modules.ts";

// Menü kurucu — deny-by-default. .NET MenuBuilder ile aynı sonuç.
export function buildMenu(session: Session): ModuleDef[] {
  return APP_MODULES.filter((m) => canSeeMenu(session, m.key));
}
